webapp/lib/controller_view_helper.dart

```diff
@@ -152,3 +152,15 @@ Map<String, List<model.SuggestedReply>> _groupRepliesIntoCategories(List<model.S
   }
   return result;
 }
+
+Map<String, List<model.Tag>> _groupTagsIntoCategories(List<model.Tag> tags) {
+  Map<String, List<model.Tag>> result = {};
+  for (model.Tag tag in tags) {
+    String category = tag.category ?? '';
+    if (!result.containsKey(category)) {
```

Comment:
Another option... `result.putIfAbsent(category, () => []).add(tag);`
Only involves 1 hash map lookup rather than 2.

Reply:
So much better, I don't know why I keep forgetting that `putIfAbsent` exists! Thanks @danrubel!

I'll update the similar code for suggested replies as well in a future PR